### DISCUSSION TITLE
Authentication for JS, Kotlin and Swift

### DISCUSSION THREAD

**Discussion by dngferreira** [sqlitecloud/discussions#13] [Help] [answered]
Hello,
To start I'm really interested in using these solution and it looks like a promising idea.
I have some things I don't think are talked about on the docs (maybe I missed it) and that leave me a bit concerned when hear about this distributed solutions which sync to a cloud.
What is the recommended way to manage the access to sqlite cloud on client side applications in js, swift or kotlin? 
It's tricky to keep secrets safe in those situations. The apps can always be messed with and get the creds out.
Should we map app users to sqlite cloud users?

Just curious on how to proceed in these scenarios.

**Comment by btholt** [comment, marked as answer]
The precise design on how this going to work is still a work in progress but the general gist is that via sharding you'll only allow client users to be able to access and/or write data they have permission to. Therefore they could mess with their own data (just like they could with API calls) but wouldn't be able to access or overwrite data they don't have access to.

If you have more sensitive situations than that, you can just keep using the traditional client/server model and handle all of it on the server.
